allennlp/data/instance.py

```diff
@@ -22,7 +22,7 @@ class Instance:
     fields : ``Dict[str, Field]``
         The ``Field`` objects that will be used to produce data arrays for this instance.
     """
-    def __init__(self, fields: Mapping[str, Field]) -> None:
+    def __init__(self, fields: MutableMapping[str, Field]) -> None:
```

Comment:
What does using `MutableMapping` give you over just using `Dict`?  Seems like it adds some complexity that we don't really need?

Reply:
https://github.com/allenai/allennlp/pull/725/files#r164889705 😇

Reply:
and I had to change it to mutable because the "add epoch" functionality creates a new field

Reply:
Ah, thanks for the reminder =).